src/tls_stream.rs

```diff
@@ -60,6 +52,17 @@ impl Builder {
         self
     }
 
+    /// Set a verification callback to be used for connections created with this `Builder`.
+    ///
+    /// The callback is provided with a boolean indicating if the (pre)validation was
+    /// successful and with the certificate, which was validated.
+    pub fn verify_callback<F>(&mut self, callback: F) -> &mut Builder 
+        where F: Fn(bool, &CertChain) -> bool + 'static
```

Comment:
I think we may want to swap the `bool`s out for `io::Result<()>` - it allows the closure to inspect the upstream validation error and pick which error they want to produce.